tumblr-photo-video-ripper.py

```diff
@@ -68,8 +68,23 @@ def run(self):
     def download(self, medium_type, post, target_folder):
         try:
             medium_url = self._handle_medium_url(medium_type, post)
-            if medium_url is not None:
-                self._download(medium_type, medium_url, target_folder)
+            #print("medium url is %s", medium_url)
+            resp_raw = requests.get(medium_url, stream=True, proxies=self.proxies, timeout=TIMEOUT)
+            if medium_type == "video":
+                self._download(medium_type, medium_url, target_folder, resp_raw)
+            elif medium_type == "photo":
+                medium_url_bak = medium_url
+                medium_url_dot = medium_url.split('.')
```

Comment:
The url parsing here seems complex and error-prone.

Below part is a better way. WDYT?
```
    def download(self, medium_type, post, target_folder):
        try:
            medium_url = self._handle_medium_url(medium_type, post)
            if medium_url is not None:
                if medium_type == "photo":
                    try:
                        # try to download raw image
                        medium_url_raw = medium_url.replace("68.media.tumblr.com", "data.tumblr.com")
                        raw_matched = self.hd_photo_regex.match(medium_url_raw)
                        if raw_matched is not None:
                            replace_raw = raw_matched.groups()[0]
                            replace_raw = replace_raw.replace(raw_matched.groups()[1], "raw")
                            medium_url_raw = medium_url_raw.replace(raw_matched.groups()[0], replace_raw)
                            self._download(medium_type, medium_url_raw, target_folder)
                            return
                    except:
                        pass
                self._download(medium_type, medium_url, target_folder)
        except TypeError:
            pass

    # can register differnet regex match rules
    def _register_regex_match_rules(self):
        # will iterate all the rules
        # the first matched result will be returned
        self.regex_rules = [video_hd_match(), video_default_match()]
        self.hd_photo_regex = re.compile(r".*(tumblr_\w+_(\d+))", re.IGNORECASE)
```